core/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java

```diff
@@ -240,10 +236,10 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeOptionalWriteable(storedFieldsContext);
         out.writeVInt(from);
         out.writeOptionalWriteable(highlightBuilder);
-        int indexBoostSize = indexBoost == null ? 0 : indexBoost.size();
-        out.writeVInt(indexBoostSize);
-        if (indexBoostSize > 0) {
-            writeIndexBoost(out);
+        if (indexBoosts == null) {
```

Comment:
Should we initialize indexBoosts to new ArrayList<>() straight-away instead and remove this null invariant? This if would go away too.